packages/core/src/services/providerApiKeys/helpers/validateName.ts

```diff
@@ -0,0 +1,32 @@
+import { Result } from '../../../lib/Result'
+import { BadRequestError } from '../../../lib/errors'
+import { ProviderApiKeysRepository } from '../../../repositories/providerApiKeysRepository'
+import Transaction from '../../../lib/Transaction'
+
+export async function validateProviderApiKeyName(
+  {
+    name,
+    workspaceId,
+  }: {
+    name: string
+    workspaceId: number
+  },
+  transaction = new Transaction(),
```

Comment:
services that only perform read operations can receive the database client instead of a transaction instance